### PR TITLE
Enhance Serializer Interface - Filename Logic

### PR DESCRIPTION
**Pull request by mbtools**
Using the enhanced serializer interface, the filename logic now uses `zcl_abapgit_object_devc` to determine the filename for packages.

### CODE REVIEW

**Comment by larshp** [comment]
I was fixing transpiler bugs exactly in the direction of the error yesterday, lets try merging https://github.com/abapGit/abapGit/pull/6209, if it doesnt work after that I'll investigate

**Comment by mbtools** [comment]
unfortunately still failing in ajson

File: test/abap_transpile.json [hunk 1]
```diff
@@ -95,6 +95,9 @@
       {"object": "ZCL_ABAPGIT_FILE_STATUS", "class": "ltcl_calculate_status", "method": "deleted_remote", "note": "READ TABLE subrc 4 vs subrc 8"},
       {"object": "ZCL_ABAPGIT_FILE_STATUS", "class": "ltcl_calculate_status", "method": "complete", "note": "READ TABLE subrc 4 vs subrc 8"},
 
+      {"object": "ZCL_ABAPGIT_FILENAME_LOGIC", "class": "ltcl_run_checks", "method": "file_to_object_package", "note": "Depends on ZCL_ABAPGIT_OBJECT_DEVC, db access"},
+      {"object": "ZCL_ABAPGIT_FILENAME_LOGIC", "class": "ltcl_run_checks", "method": "object_to_file_package", "note": "Depends on ZCL_ABAPGIT_OBJECT_DEVC, db access"},
+
```
Comment (larshp) [inline test/abap_transpile.json:100]: hmm, these unit tests worked before the change? does the tests pass when running in a real system? and there is no $PACK in the system?

Comment (larshp) [reply test/abap_transpile.json]: fixed with https://github.com/abapGit/abapGit/pull/6235/commits/b6304aa9c7f18e2436700924ea9d2b1f863612f2

**Review by larshp — APPROVED** [review]

**Pull request merged**